src/PIDController.cpp

```diff
@@ -0,0 +1,12 @@
+#include "PIDController.h"
+
+PIDController::PIDController(double p, double i, double d) {
+    p_, i_, d_ = p, i, d;
+    lastUpdate_ = micros();
+    previousError_ = 0;
+}
+
+double PIDController::update(double error) {
+    float dt = micros() - lastUpdate_;
+    return p_*error+d_*(error-previousError_)/dt;
```

Comment:
no integral term?
Also, calculating derivative using 2 previous terms is very noisy. It's not the most legit method, but consider using the old algorithm, which loads datapoints into a buffer and running linear regression.